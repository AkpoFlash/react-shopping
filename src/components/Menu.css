.card__item{
  width: 600px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-align: center;
}

.card__button{
  width: 100%;
  height: 35px;
}

.card__count{
  font-weight: bold;
  margin: 5px 0;
}
