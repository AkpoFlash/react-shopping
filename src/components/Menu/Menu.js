import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { t } from '../../helpers';
import Basket from '../Basket/Basket';

const StyledMenu = styled.nav`
	width: 100%;
	grid-column: 2;
	display: flex;
  justify-content: space-between;
  
	& ul {
		list-style: none;
		padding: 0;
		margin: 0;
		height: 100%;

		& li {
			text-align: center;
			box-sizing: border-box;
			display: inline-block;
			padding: 5px 15px;
			height: 100%;
		}
  }
`;

const Menu = (props) => {

  const handleItemClick = (e) => {}

  return(
    <StyledMenu>
      <ul>
        <li name='browse' onClick={handleItemClick}>
          {t('Books shop')}
        </li>
        <li name='browse' onClick={handleItemClick}>
          {t('Books shop')}
        </li>
      </ul>
        

      <ul>
        <li>
          {t('Basket is empty')}
          {/* <Popup
            trigger={
                <li name='help' onClick={ handleItemClick }>
                  {t('Basket')}: { props.totalPrice } {t('Currency')} (<b>{ props.count }</b>)
                </li>
              }
            content={
              props.count
              ?  <Basket { ...props } />
            : <div>{t('Basket is empty')}</div>
            }
            on='click'
            hideOnScroll
          /> */}
        </li>
      </ul>
    </StyledMenu>
  );
}

const mapStateToProps = (state) => ({
  totalPrice: state.card.items.reduce( (total, book) => total + book.price, 0),
  count: state.card.items.length,
  items: state.card.items,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
