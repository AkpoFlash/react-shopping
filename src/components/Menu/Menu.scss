.menu {
	width: 100%;
	grid-column: 2;
	display: flex;
	justify-content: space-between;

	& ul {
		list-style: none;
		padding: 0;
		margin: 0;
		height: 100%;

		& li {
			text-align: center;
			box-sizing: border-box;
			display: inline-block;
			padding: 5px 15px;
			height: 100%;
		}
	}

}

.card__item{
  width: 600px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-align: center;
}

.card__button{
  width: 100%;
  height: 35px;
}

.card__count{
  font-weight: bold;
  margin: 5px 0;
}
