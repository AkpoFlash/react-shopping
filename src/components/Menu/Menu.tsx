import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import * as PropTypes from 'prop-types';
import * as moment from 'moment';

import t from '~/helpers/translator';
import Basket from '~/components/Basket/Basket';
import LangSelect from '~/components/LangSelect/LangSelect';
import { CARD_TYPE, LANGUAGES_TYPE } from '~/constants/types';

const StyledMenu = styled.nav`
	width: 100%;
	grid-column: 2;
	display: flex;
  justify-content: space-between;
  
	& ul {
		list-style: none;
		padding: 0;
		margin: 0;
		height: 100%;

		& li {
			text-align: center;
			box-sizing: border-box;
			display: inline-block;
			padding: 5px 15px;
			height: 100%;
      vertical-align: top;
		}
  }
`;

const SubTitle = styled.div`
  font-size: 0.75rem;
`;

interface Props {
  card: CARD_TYPE;
  languages: LANGUAGES_TYPE;
  count: number,
  totalPrice: number,
  usersLang: string,
}

export const Menu = (props: Props) => {

  // Change
  moment.locale(props.usersLang.slice(0,-1));
  const dateCouldOrderGive = moment().add(7, 'd').startOf('d').add(12, 'h').fromNow();

  const handleItemClick = () => {
    /* TODO setState({ showBasket: !this.state.showBasket }); */
  }

  return (
    <StyledMenu>
      <ul>
        <NavLink to='/'>
          <li>
            {t('Home')}
          </li>
        </NavLink>
        <NavLink to='/books'>
          <li>
            {t('Books shop')}
          </li>
        </NavLink>
      </ul>
      <ul>
        <li>
          <LangSelect />
        </li>
        <li>
          { props.count ?
            <li onClick={ handleItemClick }>
              {t('Basket')}: {props.totalPrice} {t('Currency')} (<b>{props.count}</b>)
              <SubTitle>
                {`${t('You can give book:')} ${dateCouldOrderGive}`}
              </SubTitle>
            </li>
            // <Basket {...props} />
            :
            t('Basket is empty')
          }
          {/* {t('Basket is empty')}
          <Popup
            trigger={
                <li name='help' onClick={ handleItemClick }>
                  {t('Basket')}: { props.totalPrice } {t('Currency')} (<b>{ props.count }</b>)
                </li>
              }
            content={
              props.count
              ?  <Basket { ...props } />
            : <div>{t('Basket is empty')}</div>
            }
            on='click'
            hideOnScroll
          /> */}
        </li>
      </ul>
    </StyledMenu>
  );
};

const mapStateToProps = ({ card, languages }: Props) => ({
  totalPrice: card.items.reduce( (total, book) => total + book.price, 0),
  count: card.items.length,
  items: card.items,
  usersLang: languages.usersLang,
});

Menu.propTypes = {
  totalPrice: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  items: PropTypes.array.isRequired,
  usersLang: PropTypes.string.isRequired,
};

export default connect(mapStateToProps)(React.memo(Menu));
