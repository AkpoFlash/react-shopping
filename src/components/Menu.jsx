import React from 'react';
import { Menu as MenuBase, Popup } from 'semantic-ui-react';

import Basket from './Basket';

import './Menu.css';

const Menu = (props) => {

  const handleItemClick = (e) => {}

  return(
    <MenuBase>
      <MenuBase.Item name='browse' onClick={ handleItemClick }>
        Магазин книг
      </MenuBase.Item>

      <MenuBase.Menu position='right'>

        <Popup
          trigger={
              <MenuBase.Item name='help' onClick={ handleItemClick }>
                Корзина: { props.totalPrice } руб. (<b>{ props.count }</b>)
              </MenuBase.Item>
            }
          content={
            props.count
            ?  <Basket { ...props } />
            : <div>Ваша корзина пуста</div>
          }
          on='click'
          hideOnScroll
        />
      </MenuBase.Menu>
    </MenuBase>
  );
}

export default Menu;
