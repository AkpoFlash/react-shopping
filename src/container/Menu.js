import { connect } from 'react-redux';
import Menu from '../components/Menu';
//import uniqBy from 'lodash/uniqBy';

const mapStateToProps = (state) => ({
  totalPrice: state.card.items.reduce( (total, book) => total + book.price, 0),
  count: state.card.items.length,
  items: state.card.items,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
