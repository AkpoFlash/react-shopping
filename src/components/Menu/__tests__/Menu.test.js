import React from 'react';

import { Menu } from '../Menu';

describe('Dumb Menu', () => {
	const menu = shallow(<Menu usersLang='En' />);

	it('check Menu snapshot', () => {
		expect(menu).toMatchSnapshot();
	});
});