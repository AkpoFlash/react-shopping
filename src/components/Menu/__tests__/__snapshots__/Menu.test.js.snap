// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dumb Menu check Menu snapshot 1`] = `
<styled.nav>
  <ul>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    >
      <li>
        Home
      </li>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/books"
    >
      <li>
        Books
      </li>
    </NavLink>
  </ul>
  <ul>
    <li>
      <Connect(Component) />
    </li>
    <li>
      <React.Memo />
    </li>
  </ul>
</styled.nav>
`;
